src/main/java/lotto/domain/LottoResult.java

```diff
@@ -0,0 +1,60 @@
+package lotto.domain;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.joining;
+
+public class LottoResult {
+    private static final int INITIAL_NUMBER = 0;
+    private static final int DECIMAL_POINT = 10;
+    private static final int PERCENTAGE = 100;
+    private final Map<Rank, Integer> lottoResult = new LinkedHashMap<>();
+
+    public LottoResult() {
+        for (Rank rank : Rank.values()) {
+            lottoResult.put(rank, INITIAL_NUMBER);
+        }
+    }
+
+    public void addRank(Rank rank) {
+        lottoResult.put(rank, lottoResult.get(rank) + 1);
+    }
+
+    public double calculateProfit(Money money) {
+        return (((double) (Math.round((earningPrice() / money.getMoney()) * DECIMAL_POINT * PERCENTAGE)))
+                / DECIMAL_POINT);
+    }
+
+    private double earningPrice() {
+        return lottoResult.entrySet()
+                .stream()
+                .map(this::calculatePrice)
+                .reduce(0.0, Double::sum);
```

Comment:
```suggestion
                .stream()
                .mapToDouble(this::calculatePrice)
                .sum()
```

근데 잘 모르겠어요. 왜 더블로 반환하는 건가요??

Reply:
mapToDouble 쓸 생각을 못했네요...!

저도 처음에 자료형은 int로 했었는데 최대값을 고려해서 double 로 바꿨습니다!
제가 설정한 로또 최대구입금액 10만원 -> 100장
100장 모두 1등시 100*20억 = 2000억
2000억을 int 로 반환할려면 오버플로가 나서 double 로 반환했습니다!